pkg/plugin/velero/item_snapshotter/v1/item_snapshotter.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"time"
+
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+)
+
+type AlsoHandlesInput struct {
+	ItemToSnapshot runtime.Unstructured
+	Backup         *api.Backup
+}
+
+type SnapshotItemInput struct {
+	Item   runtime.Unstructured
+	Params map[string]string
+	Backup *api.Backup
+}
+
+type SnapshotItemOutput struct {
+	UpdatedItem      runtime.Unstructured
+	SnapshotID       string
+	SnapshotMetadata map[string]string
```

Comment:
could you give a concrete example of `SnapshotMetadata`?  This is not mentioned in the design and have you considered adding a `GetSnapshot` method to return the detail information of the snapshot?